tests/providers/common/audit_info_test.py

```diff
@@ -198,3 +210,40 @@ def test_set_audit_info_azure(self):
 
         audit_info = set_provider_audit_info(provider, arguments)
         assert isinstance(audit_info, Azure_Audit_Info)
+
+    @mock_resourcegroupstaggingapi
+    @mock_ec2
+    def test_get_tagged_resources(self):
+        client = boto3.client("ec2", region_name="eu-central-1")
+        instances = client.run_instances(
+            ImageId=EXAMPLE_AMI_ID,
+            MinCount=1,
+            MaxCount=1,
+            InstanceType="t2.micro",
+            TagSpecifications=[
+                {
+                    "ResourceType": "instance",
+                    "Tags": [
+                        {"Key": "MY_TAG1", "Value": "MY_VALUE1"},
+                        {"Key": "MY_TAG2", "Value": "MY_VALUE2"},
+                    ],
+                },
+                {
+                    "ResourceType": "instance",
+                    "Tags": [{"Key": "ami", "Value": "test"}],
+                },
+            ],
+        )
+        instance_id = instances["Instances"][0]["InstanceId"]
+        image_id = client.create_image(Name="testami", InstanceId=instance_id)[
+            "ImageId"
+        ]
+        client.create_tags(Resources=[image_id], Tags=[{"Key": "ami", "Value": "test"}])
+
+        mock_current_audit_info.audited_regions = ["eu-central-1"]
+        mock_current_audit_info.audit_session = boto3.session.Session()
+        assert len(get_tagged_resources(["ami=test"], mock_current_audit_info)) == 2
+        assert (
+            len(get_tagged_resources(["MY_TAG1=MY_VALUE1"], mock_current_audit_info))
+            == 1
+        )
```

Comment:
Can you split this into two separate test? Thanks!

Reply:
I don't feel it is necessary, it is a same environment for different input tags. I have added more asserts though.

Reply:
Cool